pkg/utils/url.go

```diff
@@ -31,3 +31,14 @@ func UnescapeURLArgument(r *http.Request, key string) (string, error) {
 
 	return url, nil
 }
+
+// IsSubdomainOf checks whether `url` is a subdomain of `parent`
+func IsSubdomainOf(url *url.URL, parent string) bool {
```

Comment:
I decided to only use this in my own implementation for now. Migrating other resolvers to this function should happen in  a separate PR.